python/ray/rllib/optimizers/policy_optimizer.py

```diff
@@ -89,8 +89,8 @@ def collect_metrics(self, min_history=100):
             res (dict): A training result dict from evaluator metrics with
                 `info` replaced with stats from self.
         """
-        episodes = collect_episodes(self.local_evaluator,
-                                    self.remote_evaluators)
+        episodes, num_dropped = collect_episodes(self.local_evaluator,
+                                                 self.remote_evaluators)
```

Comment:
note that we need to process num_dropped here as well

Reply:
I added `num_dropped` to the returned dict in the following lines actually. Anyway, I will move the stuffs into `summarize_episodes()` as you showed.